### DISCUSSION TITLE
How do I shutdown/exit gotty server once my command completes?

### DISCUSSION THREAD

**Discussion by mrinalwahal** [sorenisanerd/gotty#62] [Q&A]
For example, if I'm running vim with `gotty -w vi test.txt` - how do I make sure gotty shuts down after I close vim?

Or basically finish any command with ctrl+c?

@sorenisanerd

**Comment by sorenisanerd** [comment]
Is the `--once` option perhaps what you are looking for?

**Reply by mrinalwahal** [reply]
Nope. For some reason when I use the `--once` option the client disconnects even before I can access my command output.

For some reason, gotty needs at least 2 connections before letting me access my command output.

Maybe because I'm running gotty behind an nginx ingress.

Is there no scriptable solution? Maybe something I can add in the command itself to shutdown gotty when I send a SIGINT (ctrl+c)?

**Reply by sorenisanerd** [reply]
Sure, you could certainly kill gotty from your terminal session, but what you're describing is exactly what the `--once` option is meant for, so I'd love to get that fixed.

Can you show me the output from Gotty when this happens and maybe the logs from nginx? Depending on your setup, nginx might be doing health checks or perhaps there's something happening when the HTTP connection gets upgraded to a websocket connection. I'm not sure, but let's get it fixed either way :)

**Reply by mrinalwahal** [reply]
I can't reproduce the logs right now. 

Can you please tell me how can I kill gotty using a SIGINT (ctrl+c)? That will solve my problem.

**Reply by sorenisanerd** [reply]
Just Ctrl-C won't work, because your session gets its own controlling tty, which is what determines where the SIGINT gets sent.

`killall gotty` would kill all gotty processes. Maybe that's useful. Depending on your shell, you might be able to have it call that when you close your session. bash, for instance, could do it using something like `trap 'killall gotty' INT EXIT TERM`.

I'm on my phone, so I can't easily test it. Let me know how it goes :)
